components/brave_wallet/common/brave_wallet.mojom

```diff
@@ -455,67 +455,61 @@ union SwapQuoteUnion {
 
 union SwapTransactionUnion {
   string jupiter_transaction;
-  ZeroExQuote zero_ex_transaction;
+  ZeroExTransaction zero_ex_transaction;
   LiFiTransactionUnion lifi_transaction;
   SquidTransactionUnion squid_transaction;
 };
 
-struct ZeroExSource {
-  string name;
-  string proportion;
-};
-
 struct ZeroExFee {
-  string fee_type;
-  string fee_token;
-  string fee_amount;
-  string billing_type;
+  string amount;
+  string token;
+  string type;
 };
 
 struct ZeroExFees {
   ZeroExFee? zero_ex_fee;
 };
 
-struct ZeroExQuote {
-  string price;
```

Comment:
is it possible to keep the `price` in `ZeroExQuote`? we are using it to calculate sell_token's market value.

Reply:
Is it possible to infer this on this client-side like we do on desktop? See https://github.com/brave/brave-core/blob/369f9d6f9804cd57e6e3df9e9891d852f3c82706/components/brave_wallet_ui/page/screens/swap/swap.utils.ts#L83-L85

In the backend, we no longer have this field and we don't know about the buy/sell token decimals to be able to correctly compute the rate.

Reply:
we have no problem with sell amount and buy amount. just the market price for the sell token.
i think iOS can do one of the following:
1. fetch market price for sell token using `assetRatioService` api instead of from quote response.
2. remove this market price label but need to move that refresh button else where for users to refetch quote. 
cc @jamesmudgett and design @kleantzogu

Reply:
My understanding is that "Market Price in ETH" is the exchange rate between `buyToken` and `sellToken`. I don't think using the `assetRatioService` will be correct here, since it's the CEX/global exchange rate and not the one as per the AMM swap quote.

Reply:
you are right. i think this leave us removing that label. i just need to figure out where to put that refetch button.

Reply:
Highlighting again (see https://github.com/brave/brave-core/pull/26231#discussion_r1828738779) that the "Market price in &lt;sellToken&gt;" can be determined using `buyAmount` and `sellAmount` formatted with their respective decimals.

Reply:
Note that it's "Market price **in** ETH" and not "Market price of ETH".

Reply:
ah! i mistakenly thought the sell amount and buy amount are from the `ZeroExTransaction`. ok yeah i think we can using those two values to determine the market price. Thanks!